Core/Core/Extensions/URLExtensions.swift

```diff
@@ -28,58 +28,64 @@ extension Sequence where Element == URL {
     }
 }
 
-extension URL {
-    public static var temporaryDirectory: URL {
-        return URL(fileURLWithPath: NSTemporaryDirectory())
-    }
+public extension URL {
+    struct Directories {
```

Comment:
Let's use a caseless enum for namespaces like Apple does. That way we don't need the private init.